File: core/src/main/scala/stasis/core/discovery/ServiceApiClient.scala
package stasis.core.discovery

trait ServiceApiClient

object ServiceApiClient {
  trait Factory {
    def create(endpoint: ServiceApiEndpoint.Api, coreClient: ServiceApiClient): ServiceApiClient
    def create(endpoint: ServiceApiEndpoint.Core): ServiceApiClient
    def create(endpoint: ServiceApiEndpoint.Discovery): ServiceApiClient
  }
}

File: client-android/lib/src/main/kotlin/stasis/client_android/lib/discovery/ServiceApiClient.kt
package stasis.client_android.lib.discovery

interface ServiceApiClient {
    interface Factory {
        fun create(endpoint: ServiceApiEndpoint.Api, coreClient: ServiceApiClient): ServiceApiClient
        fun create(endpoint: ServiceApiEndpoint.Core): ServiceApiClient
        fun create(endpoint: ServiceApiEndpoint.Discovery): ServiceApiClient
    }
}
